### ISSUE TITLE
Table MaskedColumn updating issues

### ISSUE THREAD

**Issue by cdeil** [astropy/astropy#2997]
@taldcroft Can you please look at example 1 and 3 in this notebook?
http://nbviewer.ipython.org/gist/cdeil/69b79c04b2477dbf611a

Is this a bug in the masked table implementation?

If not, can you point me to or add documentation how one can update missing data in Astropy tables?
I did look through the docs a bit, but couldn't find it.

Naively this is what I tried ... it results in a table where the mask is in a weird (corrupt?) state:

``` python
# Try to update missing information in column `b` from column `c`
mask = t['b'].mask
t['b'][mask] = t['c'][mask]
```

While looking through the docs I saw [this section](http://astropy.readthedocs.org/en/latest/table/masking.html#table-access) which explained that rows are views for non-masked tables, but copies for masked tables. That's another bug I had in my script. Aargh ... I'm thinking of re-writing my script completely using `pandas.DataFrame` instead, in the hope that it will  let me process my data with missing values more easily.

**Comment by cdeil** [comment]
I re-ran the notebook that shows the bug again with Astropy master:
http://nbviewer.jupyter.org/gist/cdeil/a33b8925fe6b0f9fba410175a3ccbf18

As far as I can see, the behaviour and results are OK not, and this issue can be closed.
@taldcroft or @mhvk - Can you please double-check?

The one thing is that there's this warning:
```
/Users/deil/Library/Python/3.5/lib/python/site-packages/astropy/table/column.py:1096: MaskedArrayFutureWarning: setting an item on a masked array which has a shared mask will not copy the mask and also change the original mask array in the future.
Check the NumPy 1.11 release notes for more information.
  ma.MaskedArray.__setitem__(self, index, value)
```
Is it possible to change something in `astropy/table/column.py:1096` so that the code works with old / new Numpy and doesn't emit this warning?

> I marked this as fixed when #2790 gets merged, but we'll leave it open until then. I'll be sure to add tests for this case.

Is a test case in place?

**Comment by saimn** [comment]
For `MaskedArrayFutureWarning` see #5060: the behavior has not changed yet. I think it will be changed in Numpy 1.13 but I cannot find the related issue.

**Comment by mhvk** [comment]
The relevant numpy issues is https://github.com/numpy/numpy/pull/5580
Bottom line: in the future, both data and mask get shared for views. I made a new issue to keep track of that (#5530), but @saimn correctly points out that #5060 already exists. We should move discussion of the warning there.

Anyway, for *this* issue, we can ignore the warnings, which, I think, means it can be closed (I confirmed that the original example now works). So, I'll do that!
